// lj_modules/requestHandlers.js
/* 
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
http = require("http");
var sax = require('sax');

exports.journeyplanner = function(req, res){
    console.log("journey planner called");
    var path = "/api/XML_TRIP_REQUEST2?language=en&sessionID=0&place_origin=London&type_origin=locator&name_origin=SE83JU&place_destination=London&type_destination=locator&name_destination=SE13UW";
    var options = {
        hostname: 'jpapi.tfl.gov.uk',
        port: 80,
        path: path
    };
    var data ="";
    http.get(options, function(tflResponse) {
        tflResponse.on('data', function(chunk){
            data += chunk;
        
        }).on("end", function(){

            parser=new DOMParser();
            xmlDoc=parser.parseFromString(data,"text/xml");

            data = xmlDoc.getElementsByTagName("itdTripRequest")[0].childNodes[0].nodeValue;

            //          parser = sax.parser(true);
            //            parser.ontext = function(t) {
            //                console.log('This is the text in that node: ' + t);
            //            };
            //            parser.write(data.toString('utf8')).close();
            res.end(data);
            res.render('journeyplannerSuccess', {
                title: "Take me home",
                xmlData: data
            })
        });

    }).on('error', function(e) {
        console.log("Got error: " + e.message);
    });
}
